specs/light_client/merkle_proofs.md

```diff
@@ -116,72 +202,69 @@ x x . . . . x *
-    maximal_indices = set()
-    for i in tree_indices:
-        x = i
-        while x > 1:
-            maximal_indices.add(x ^ 1)
-            x //= 2
-    maximal_indices = tree_indices + sorted(list(maximal_indices))[::-1]
-    # Get indices that cannot be recalculated from earlier indices
-    redundant_indices = set()
-    proof = []
-    for index in maximal_indices:
-        if index not in redundant_indices:
-            proof.append(index)
-            while index > 1:
-                redundant_indices.add(index)
-                if (index ^ 1) not in redundant_indices:
-                    break
-                index //= 2
-    return [i for i in proof if i not in tree_indices]
-```
-
-Generating a proof is simply a matter of taking the node of the SSZ hash tree with the union of the given generalized indices for each index given by `get_proof_indices`, and outputting the list of nodes in the same order.
-
-Here is the verification function:
+First, we provide a method for computing the generalized indices of the auxiliary tree nodes that a proof of a given set of generalized indices will require:
 
 ```python
-def verify_multi_proof(root: Bytes32, indices: List[int], leaves: List[Bytes32], proof: List[Bytes32]) -> bool:
-    tree = {}
-    for index, leaf in zip(indices, leaves):
-        tree[index] = leaf
-    for index, proof_item in zip(get_proof_indices(indices), proof):
-        tree[index] = proof_item
-    index_queue = sorted(tree.keys())[:-1]
-    i = 0
-    while i < len(index_queue):
-        index = index_queue[i]
-        if index >= 2 and index ^ 1 in tree:
-            tree[index // 2] = hash(tree[index - index % 2] + tree[index - index % 2 + 1])
-            index_queue.append(index // 2)
-        i += 1
-    return (indices == []) or (1 in tree and tree[1] == root)
+def get_branch_indices(tree_index: GeneralizedIndex) -> List[GeneralizedIndex]:
+    """
+    Get the generalized indices of the sister chunks along the path from the chunk with the
+    given tree index to the root.
+    """
+    o = [generalized_index_sibling(tree_index)]
+    while o[-1] > 1:
+        o.append(generalized_index_sibling(generalized_index_parent(o[-1])))
+    return o[:-1]
+
+def get_helper_indices(indices: List[GeneralizedIndex]) -> List[GeneralizedIndex]:
+    """
+    Get the generalized indices of all "extra" chunks in the tree needed to prove the chunks with the given
+    generalized indices. Note that the decreasing order is chosen deliberately to ensure equivalence to the
+    order of hashes in a regular single-item Merkle proof in the single-item case.
+    """
+    all_indices = set()
+    for index in indices:
+        all_indices = all_indices.union(set(get_branch_indices(index) + [index]))
+    
+    return sorted([
+        x for x in all_indices if not
+        (generalized_index_child(x, 0) in all_indices and generalized_index_child(x, 1) in all_indices) and not
+        (x in indices)
+    ])[::-1]
```

Comment:
Reversal of the list may be a bit opaque here for non-python readers. Instead we can just make the `sorted` descending order.
```suggestion
    ], reverse=True)
```